src/test/java/org/jitsi/jicofo/ParticipantTest.java

```diff
@@ -44,88 +42,397 @@
+            createGroupSSRC(10L, cname, msid),
+            createGroupSSRC(20L, cname, msid),
+            createGroupSSRC(30L, cname, msid),
+            createGroupSSRC(40L, cname, msid),
+            createGroupSSRC(50L, cname, msid),
+            createGroupSSRC(60L, cname, msid)
+        };
+
+        this.videoGroups = new SourceGroupPacketExtension[] {
+            createGroup(
+                SourceGroupPacketExtension.SEMANTICS_SIMULCAST,
+                new long[] { 10L, 30L, 50L },
+                cname, msid),
+            createGroup("FIR", new long[] { 10L, 20L }, cname, msid),
+            createGroup("FIR", new long[] { 30L, 40L }, cname, msid),
+            createGroup("FIR", new long[] { 50L, 60L }, cname, msid)
+        };
 
         ContentPacketExtension audioContents
             = JingleOfferFactory.createAudioContent(false, true, true);
 
-        RtpDescriptionPacketExtension audioRtpDescPe
-            = JingleUtils.getRtpDescription(audioContents);
+        ContentPacketExtension videoContents
+            = JingleOfferFactory.createVideoContent(false, true, true, 0, 100);
+
+        this.audioRtpDescPe = JingleUtils.getRtpDescription(audioContents);
+        this.videoRtpDescPe = JingleUtils.getRtpDescription(videoContents);
+
+        this.answerContents = new ArrayList<>();
+
+        answerContents.add(audioContents);
+        answerContents.add(videoContents);
+    }
 
+    private void addDefaultAudioSSRCs()
+    {
         for (SourcePacketExtension ssrc : audioSSRCs)
         {
             audioRtpDescPe.addChildExtension(ssrc);
         }
+    }
 
-        List<ContentPacketExtension> answerContents = new ArrayList<>();
+    private void addDefaultVideoSSRCs()
+    {
+        for (SourcePacketExtension ssrc : videoSSRCs)
+            videoRtpDescPe.addChildExtension(ssrc);
+    }
 
-        answerContents.add(audioContents);
+    private void addDefaultVideoGroups()
+    {
+        for (SourceGroupPacketExtension group : videoGroups)
+            videoRtpDescPe.addChildExtension(group);
+    }
+
+    @Test
+    public void testNegative()
+    {
+        audioRtpDescPe.addChildExtension(createSSRC(-1));
+
+        this.addDefaultAudioSSRCs();
+
+        try
+        {
+            participant.addSSRCsAndGroupsFromContent(answerContents);
+            fail("Did not detect SSRC -1 as invalid");
+        }
+        catch (InvalidSSRCsException exc)
+        {
+            assertEquals("Illegal SSRC value: -1", exc.getMessage());
+        }
+    }
+
+    @Test
+    public void testZero()
+    {
+        audioRtpDescPe.addChildExtension(createSSRC(0));
+
+        this.addDefaultAudioSSRCs();
+
+        try
+        {
+            participant.addSSRCsAndGroupsFromContent(answerContents);
+            fail("Did not detect SSRC 0 as invalid");
+        }
+        catch (InvalidSSRCsException exc)
+        {
+            assertEquals("Illegal SSRC value: 0", exc.getMessage());
```

Comment:
I had the same thought initially, but then when working on the tests it turned out that it's as easy as doing copy paste in case the error message is changed. Another excuse was that no real app logic depends on that.

I would prefer to not do that, but only because of being lazy. I can spend some time on fixing it if you have strong objections against keeping it this way.